internal/version/version.go

```diff
@@ -0,0 +1,167 @@
+	CommitHash string    `json:"commitHash,omitempty"`
+	Version    SemVer    `json:"version"`
+	Release    bool      `json:"release,omitempty"`
+}
+
+// SemVer represents a semantic version
+type SemVer struct {
+	Major int64 `json:"major"`
+	Minor int64 `json:"minor"`
+	Patch int64 `json:"patch"`
+}
+
+// AllowedVersions provides a list of SemVer per Service
+type AllowedVersions struct {
+	Bootstrap   []SemVer
+	Satellite   []SemVer
+	Storagenode []SemVer
+	Uplink      []SemVer
+	Gateway     []SemVer
+}
+
+// SemVerRegex is the regular expression used to parse a semantic version.
+// https://github.com/Masterminds/semver/blob/master/LICENSE.txt
+const SemVerRegex string = `v?([0-9]+)\.([0-9]+)\.([0-9]+)`
+
+// NewSemVer parses a given version and returns an instance of SemVer or
+// an error if unable to parse the version.
+func NewSemVer(regex *regexp.Regexp, v string) (*SemVer, error) {
+	m := regex.FindStringSubmatch(v)
+	if m == nil {
+		return nil, errors.New("invalid semantic version for build")
+	}
+
+	sv := SemVer{}
+
+	var err error
+
+	// first entry of m is the entire version string
+	sv.Major, err = strconv.ParseInt(m[1], 10, 64)
+	if err != nil {
+		return nil, err
+	}
+
+	if m[2] == "" {
+		sv.Minor = 0
+	} else {
+		sv.Minor, err = strconv.ParseInt(m[2], 10, 64)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if m[3] == "" {
+		sv.Patch = 0
+	} else {
+		sv.Patch, err = strconv.ParseInt(m[3], 10, 64)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return &sv, nil
+}
+
+// String converts the SemVer struct to a more easy to handle string
+func (sem *SemVer) String() (version string) {
+	return fmt.Sprintf("v%d.%d.%d", sem.Major, sem.Minor, sem.Patch)
+}
+
+// New creates Version_Info from a json byte array
+func New(data []byte) (v Info, err error) {
+	err = json.Unmarshal(data, &v)
+	return v, err
+}
+
+// Marshal converts the existing Version Info to any json byte array
+func (v Info) Marshal() (data []byte, err error) {
+	data, err = json.Marshal(v)
+	return
+}
+
+// containsVersion compares the allowed version array against the passed version
+func containsVersion(all []SemVer, x SemVer) bool {
+	for _, n := range all {
+		if x == n {
+			return true
+		}
+	}
+	return false
+}
+
+// StrToSemVerList converts a list of versions to a list of SemVer
+func StrToSemVerList(serviceVersions []string) (versions []SemVer, err error) {
+
+	versionRegex := regexp.MustCompile("^" + SemVerRegex + "$")
+
+	for _, subversion := range serviceVersions {
+		sVer, err := NewSemVer(versionRegex, subversion)
+		if err != nil {
+			return nil, err
+		}
+		versions = append(versions, *sVer)
+	}
+	return versions, err
+}
+
+func init() {
+	if buildVersion == "" || buildTimestamp == "" || buildCommitHash == "" {
+		return
+	}
+	timestamp, err := strconv.ParseInt(buildTimestamp, 10, 64)
+	if err != nil {
+		return
+	}
+	Build = Info{
+		Timestamp:  time.Unix(timestamp, 0),
+		CommitHash: buildCommitHash,
+		Release:    buildRelease,
```

Comment:
`Release: strings.ToLower(buildRelease) == "true",`

Reply:
buildRelease is a bool

Reply:
it should not be. linker flags can only set strings

Reply:
Can linker flags be private ?

Reply:
yep!